### PR TITLE
Corrections for Embedded samples 

### PR DESCRIPTION
**Pull request by harrypuuter**
This PR adds functions for the readout of corrections for embedded samples

In addition, a small for fix quantities in the global space to be included in all scopes is applied 

Related to https://github.com/KIT-CMS/TauAnalysis-CROWN/pull/3

### CODE REVIEW

File: src/scalefactors.cxx [hunk 1]
```diff
@@ -762,5 +762,163 @@ btagSF(ROOT::RDF::RNode df, const std::string &pt, const std::string &eta,
...
+ * corresponds to the leading muon selected by the embedding selection
+ * @param eta_1 the eta of the leading generator particle in the event. This
+ * corresponds to the leading muon selected by the embedding selection
+ * @param pt_2 the pt of the subleading generator particle in the event. This
+ * corresponds to the subleading muon selected by the embedding selection
+ * @param eta_2 the eta of the subleading generator particle in the event. This
+ * corresponds to the subleading muon selected by the embedding selection
+ * @param output name of the output column
+ * @param sf_file path to the correctionlib file containing the scale factor
+ * @param idAlgorithm name of the scale factor with in the correctionlib file
```
Comment (ralfschmieder) [inline src/scalefactors.cxx:781]: "in" instead of "with in"

File: src/scalefactors.cxx [hunk 2]
```diff
@@ -762,5 +762,163 @@ btagSF(ROOT::RDF::RNode df, const std::string &pt, const std::string &eta,
...
+ * @brief Function used to readout the embedding selection trigger scalefactors.
+ *
+ * @param df the input dataframe
+ * @param pt the pt of the generator particle in the event. This corresponds to
+ * one of the muons selected by the embedding selection
+ * @param eta the eta of the generator particle in the event. This corresponds
+ * to one of the muons selected by the embedding selection
+ * @param output the name of the output column
+ * @param sf_file the path to the correctionlib file containing the scale factor
+ * @param idAlgorithm the name of the scale factor with in the correctionlib
```
Comment (ralfschmieder) [inline src/scalefactors.cxx:820]: s.o.

File: src/scalefactors.cxx [hunk 3]
```diff
@@ -762,5 +762,163 @@ btagSF(ROOT::RDF::RNode df, const std::string &pt, const std::string &eta,
...
+ * @brief Function used to readout SFs from the muon scale factor measurements
+ *
+ * @param df the input dataframe
+ * @param pt the pt of the muon
+ * @param eta the eta of the muon
+ * @param output the name of the output column
+ * @param sf_file the path to the correctionlib file containing the scale factor
+ * @param correctiontype the type of the correction. Use `emb` for embedding and
+ * `mc` for monte carlo
+ * @param idAlgorithm the name of the scale factor with in the correctionlib
```
Comment (ralfschmieder) [inline src/scalefactors.cxx:856]: s.o.

**Review by ralfschmieder — APPROVED** [review]

**Comment by harrypuuter** [comment]
Resolves #163

**Pull request merged**